google/ggrc-core
# GGRC-5293/GGRC-5294/GGRC-5297/GGRC-5380 Create/update issue in issue tracker when Issue created/updated/deleted in GGRC

# Issue description

This PR contains functionality for Issue object integration from Issue Tracker side to GGRC.
It contains next main parts:
* Small assessment integration functionality refactoring (moving some functions and tests for assessment integration in separate modules);
* Basic framework for Issue tracker integration (`IssueTrackerParamsBuilder` which helps to build request to issue tracker and `IssueTrackerParamsContainer` which helps to contains issue tracker parameters and represent those parameters for params builder and for front-end);
* Common event handlers for Issue Tracker integration functionality (see _Solution description_ section for more details) and functionality for definition event handlers in declarative notation;
* Create/Update/Delete event handlers for Issue object integration.

# Steps to test the changes

Perform next actions on test env:

> Remember that `Global Auditor` can't create issue in GGRC with enabled synchronization 

 Action in GGRC | What should happen in GGRC | What should happen in Issue Tracker 
 :---------------:  | :------------------------------: | :-------------------------------------: 
| Create issue with `disabled` integration | `Issue` object should be created<br/> `IssuetrackerIssue` object shouldn't be created | -- |
| Create issue with `enabled` integration | `Issue` object should be created<br/>`IssuetrackerIssue` object should be created with info sent by front-end | Ticket should be created in Issue Tracker |
| Update issue fields with `disabled` integration | `Issue` object should be updated | -- |
| Update issue fields with `enabled` integration | `Issue` object should be updated<br/>`IssuetrackerIssue` should be updated | Comments should be added to ticket or/and fields in Issue Tracker should be updated |
| Delete issue with `disabled` integration | `Issue` object should be deleted<br/>`IssuetrackerIssue` object should be deleted if exists | -- |
| Delete issue with `enabled` integration | `Issue` object should be deleted<br/>`IssuetrackerIssue` object should be deleted | Commend should be added to ticket and ticket status should be changed |
| Turn off integration on already existing object | `IssuetrackerIssue` object should be updated | Comment should be added to ticket |
| Turn on integration on already existing object | `IssuetrackerIssue` object should be created if doesn't exist | Ticket should be created if doesn't exists<br/>In other case comment should be added to ticked | 
| Create issue with `enabled` integration being `Auditor` for any `Audit` | `Issue` should be created<br/>`IssuetrackerIssue` object shouldn't be created<br/>Warning should be shown | -- | 

# Solution description

I have implemented common event handlers for integration functionality. The main idea of this changes is containing event handlers in declarative notation. You can set `create`, `update` and `delete` handlers for models in `ISSUE_TRACKER_HANDLERS`.
There are common handlers to create, update and delete events which will call concrete handlers for concrete model.

Also this PR contains `BaseIssueTrackerParamsBuilder` and `IssueParamsBuilder`. The main idea of those classes is to contain business logic in separate class for which we can easily write unit-tests. In case of changing business we will need rewrite only params builder and handlers will not require any changes.

# Sanity checklist




# PR Review checklist
